ocaml/xapi/xapi_pusb.ml

```diff
@@ -45,7 +47,7 @@ let scan_start ~__context usbs =
                Db.PUSB.set_USB_group ~__context ~self ~value:group
   ) (USBSet.diff local_usb known_usb);
 
-  List.filter (fun (rf, rc) -> USBSet.mem (extract_known_usb_info rc) (USBSet.diff known_usb local_usb)) (Db.PUSB.get_all_records ~__context)
+  List.filter (fun (rf, rc) -> USBSet.mem (extract_known_usb_info rc) (USBSet.diff known_usb local_usb)) local_pusb_records
```

Comment:
Why do we need to go through the whole list of DB records again? We need to destroy all PUSBs that are in the set difference `known_usb \ local_usb`, or "PUSBs that are in the DB, but not found in the host". The purpose of the `List.filter` seems to find the DB refs again that were thrown away above. Can we just not do that?

More importantly, I don't see how this changes anything, because PUSBs from other hosts would not have been in `known_usb \ local_usb` anyway?

Reply:
yes, known_usb\ local_usb refer to local PUSB.  This change will indeed fix the issue. As if when use whole PUSB DB records, two exactly same PUSB records in two hosts will be filtered from this. As "rc" here will happen twice because of the same PUSBs